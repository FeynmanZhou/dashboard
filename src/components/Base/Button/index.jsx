import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Button } from 'reactstrap';
import styles from './index.scss';

export default class PiButton extends PureComponent {
  static defaultProps = {
    type: 'default'
  };

  static propTypes = {
    active: PropTypes.bool,
    block: PropTypes.bool,
    color: PropTypes.string,
    disabled: PropTypes.bool,
    tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
    onClick: PropTypes.func,
    size: PropTypes.string,
    type: PropTypes.string,        
  };

  render() {
    const { type, children } = this.props;

    let cx = classNames.bind(styles);
    let classes = cx('btn', {
      [`btn-${type}`]: type
    });

    return (
      <Button
        className={classes}
        {...this.props}
      >
        {children}
      </Button>
    );
  }
}
