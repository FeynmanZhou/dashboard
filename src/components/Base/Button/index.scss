@import "~scss/variables";

.button {
  font-family: Roboto;
  font-size: $body-size;
  font-weight: 500;
	color: #ffffff;
  line-height: 1.15;
  padding: 7px 13px 7px 14px;
  border-radius: 3px;
  cursor: pointer;

  &:focus {
    box-shadow: none;
  }
}

.btn {
  &.primary {
    @extend .button;
    border: 1px solid $primary;
    background-color: $primary;

    &:hover, &:not([disabled]):not(.disabled):active, &:disabled {
      background-color: $primary;
      border-color: $primary;  
      box-shadow: none;
    }
  }

  &.secondary {
    @extend .button;
    color: $primary;
    border: 1px solid $grey;
    background-color: #fff;

    &:hover, &:not([disabled]):not(.disabled):active, &:disabled {
      background-color: #fff;
      border-color: $grey; 
      box-shadow: none;           
    }    
  }
}
