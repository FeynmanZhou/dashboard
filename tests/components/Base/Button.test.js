import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer';

import Button from 'components/Base/Button';

configure({ adapter: new Adapter() });

describe('Base/Button', ()=> {
  it('render without crash', ()=> {
    const wrapper=shallow(<Button/>);
  });

  it('primary type', ()=> {
    const wrapper=shallow(<Button className='primary' />);
    expect(wrapper.hasClass('primary')).toBeTruthy();

    // since server-side scss is transformed, classnames may not work
    // const tree=renderer.create(<Button className='primary' />).toJSON();
    // expect(tree).toMatchSnapshot();
  });

  it('call onClick', ()=> {

  })
});