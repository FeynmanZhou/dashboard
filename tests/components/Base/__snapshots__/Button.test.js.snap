// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/Button basic render 1`] = `
<button
  class="button default loading test"
  type="submit"
>
  <i
    class="icon icon-spinner"
  />
  test button
</button>
`;
